server.py

```diff
@@ -56,11 +78,16 @@ def file_upload(number):
     expected = expected.strip()
     uploaded = uploaded.strip()
     ans = (uploaded == expected)
-    usernames[u_name].append(Submission(question=number, time=time,
+    if not u_name in usernames.keys():
+    	usernames[u_name] = User()
+    usernames[u_name].submissions.append(Submission(question=number, time=time,
                                         output=uploaded, result=ans))
     if not ans:
         return "Wrong Answer!!"
     else:
+        if not (int(number) in usernames[u_name].solvedQuestions):
+        	usernames[u_name].incrementScore()
```

Comment:
why do I have to increment AND add the solved question? Isn't one dependent on the other?

Reply:
I kept both in case we assign different sores to problems based on difficulty.

Reply:
I removed score anyways, as it was not needed as of now.